tests/server/test_resource_list.py

```diff
@@ -166,6 +169,19 @@ async def create_resource(agent: str, path: str, resource_type: str, status: Res
     await create_resource("agent2", "/tmp/dir5", "std::Directory", ResourceState.skipped, [3])
     await create_resource("agent3", "/tmp/dir6", "std::Directory", ResourceState.deployed, [3])
 
+    env2 = data.Environment(name="dev-test2", project=project.id, repo_url="", repo_branch="")
```

Comment:
This is a good idea. 

- [ ] I would also add to this env2, some resources identical to the ones from the other env, to detect that kind of leaks
- [ ] I would also make an env3 with a released version `6` and a `/etc/file3` to see if that causes it to become a non-orphan